src/Stripe.net/Services/_interfaces/IDeletable.cs

```diff
@@ -3,11 +3,12 @@ namespace Stripe
     using System.Threading;
     using System.Threading.Tasks;
 
-    public interface IDeletable<TEntity>
+    public interface IDeletable<TEntity, TOptions>
         where TEntity : IStripeEntity, IHasId
+        where TOptions : BaseOptions, new()
     {
-        TEntity Delete(string id, RequestOptions requestOptions = null);
+        TEntity Delete(string id, TOptions options, RequestOptions requestOptions = null);
```

Comment:
Mind making the argument optional with `TOptions options = null`, and doing the same thing for the async method? And update `IRetrievable`/`INestedRetrievable` to do the same thing while you're at it?

This would allow the following invocations to work:

```c#
SubscriptionItemService s1 = new SubscriptionItemService();
s1.Delete(id); // works already, because `options` is optional in `SubscriptionItemService.Delete()`

IDeletable s2 = (IDeletable) s1;
s2.Delete(id); // doesn't work currently, because `options` is not optional in `IDeletable.Delete()`
```